wemod_launcher/prepare.py

```diff
@@ -299,46 +299,38 @@ def self_update(path: List[Optional[str]]) -> List[Optional[str]]:
     return path
 
 
-def check_flatpak(flatpak_cmd: Optional[List[str]]) -> List[str]:
-    if flatpak_cmd == None:
-        if "FLATPAK_ID" in os.environ or os.path.exists("/.flatpak-info"):
-            return ["True"]
-        return []
-    else:
-        if "FLATPAK_ID" in os.environ or os.path.exists("/.flatpak-info"):
-            flatpak_start = [
-                "flatpak-spawn",
-                "--host",
-            ]
-
-            envlist = [
-                "STEAM_COMPAT_TOOL_PATHS",
-                "STEAM_COMPAT_DATA_PATH",
-                "WINE_PREFIX_PATH",
-                "WINEPREFIX",
-                "WINE",
-                "SCANFOLDER",
-                "TROUBLESHOOT",
-                "WEMOD_LOG",
-                "WAIT_ON_GAMECLOSE",
-                "SELF_UPDATE",
-                "FORCE_UPDATE_WEMOD",
-                "REPO_STRING",
-            ]
-            for env in envlist:
-                if env in os.environ:
-                    flatpak_start.append(f"--env={env}={os.environ[env]}")
-            infpr = os.getenv("WeModInfProtect", "1")
-            infpr = str(int(infpr) + 1)
-
-            flatpak_start.append("FROM_FLATPAK=true")
-            flatpak_start.append(f"WeModInfProtect={infpr}")
-            flatpak_start.append("--")  # Isolate command from command args
-
-            if bool(flatpak_cmd):  # if venv is set use it
-                flatpak_cmd = flatpak_start + flatpak_cmd
-            else:  # if not use python executable
-                flatpak_cmd = flatpak_start + [sys.executable]
+def check_flatpak():
+    if "FLATPAK_ID" in os.environ or os.path.exists("/.flatpak-info"):
+        flatpak_start = [
+            "flatpak-spawn",
+            "--host",
+        ]
+
+        envlist = [
+            "STEAM_COMPAT_TOOL_PATHS",
+            "STEAM_COMPAT_DATA_PATH",
+            "WINE_PREFIX_PATH",
+            "WINEPREFIX",
+            "WINE",
+            "SCANFOLDER",
+            "TROUBLESHOOT",
+            "WEMOD_LOG",
+            "WAIT_ON_GAMECLOSE",
+            "SELF_UPDATE",
+            "FORCE_UPDATE_WEMOD",
+            "REPO_STRING",
+        ]
+        for env in envlist:
+            if env in os.environ:
+                flatpak_start.append(f"--env={env}={os.environ[env]}")
+        infpr = os.getenv("WeModInfProtect", "1")
+        infpr = str(int(infpr) + 1)
+
+        flatpak_start.append("FROM_FLATPAK=true")
+        flatpak_start.append(f"WeModInfProtect={infpr}")
```

Comment:
i think "--env=" is missing on:

    flatpak_start.append("FROM_FLATPAK=true")
    flatpak_start.append(f"WeModInfProtect={infpr}")
        
they should be:

    flatpak_start.append("--env=FROM_FLATPAK=true")
    flatpak_start.append(f"--env=WeModInfProtect={infpr}")
    
if we use "--env=" consistency would be nice.

Reply:
Good catch. Maybe we could include it in the list?

Reply:
Resolved in 2d8d1ba.